front/app/components/SignUpIn/SignUp/index.tsx

```diff
@@ -398,11 +363,14 @@ const SignUp: FC<Props & InjectedIntlProps> = memo(
                 step={activeStep}
                 metaData={metaData}
                 onData={handleOnOutletData}
-                onSkipped={handleStepSkipped}
+                onDataLoaded={handleOnOutletDataLoaded}
                 onError={handleStepError}
-                onCompleted={handleStepCompleted}
+                onSkipped={onCompleteActiveStep}
+                onCompleted={onCompleteActiveStep}
               />
 
+              <Mounter onMount={confirmOutletsRendered} />
```

Comment:
This is the controversial solution to ensuring the `Outlet`s are rendered before calculating which step we are at etc. I think this solution is not so bad but I am also open to other options, such as adding a new function prop to the `Outlet` component that is called after it has rendered all its content.